osquery/tables/system/darwin/signature.mm

```diff
@@ -26,55 +26,55 @@
 
 namespace osquery {
 namespace tables {
-
-// Get the flags to pass to SecStaticCodeCheckValidityWithErrors, depending on
-// the OS version.
-Status getVerifyFlags(SecCSFlags& flags) {
+int getOSMinorVersion() {
   using boost::lexical_cast;
   using boost::bad_lexical_cast;
 
-  static SecCSFlags sFlags;
-
-  if (sFlags == 0) {
+  try {
     auto qd = SQL::selectAllFrom("os_version");
     if (qd.size() != 1) {
-      return Status(-1, "Couldn't determine OS X version");
+      VLOG(1) << "Couldn't determine OS X version";
+      return -1;
     }
 
-    int minorVersion;
-    try {
-      minorVersion = lexical_cast<int>(qd.front().at("minor"));
-    } catch (const bad_lexical_cast& e) {
-      return Status(-1, "Couldn't determine OS X version");
-    }
+    return lexical_cast<int>(qd.front().at("minor"));
 
-    sFlags = kSecCSStrictValidate | kSecCSCheckAllArchitectures;
-    if (minorVersion > 8) {
-      sFlags |= kSecCSCheckNestedCode;
-    }
+  } catch (const bad_lexical_cast& e) {
+    VLOG(1) << "Bad OSX minor version returned by the os_version table";
+    return -1;
+  }
+}
+
+// Get the flags to pass to SecStaticCodeCheckValidityWithErrors, depending on
+// the OS version.
+SecCSFlags getVerifyFlags(bool hash_resources) {
+  static int osMinorVersion = getOSMinorVersion();
+
+  int flags = kSecCSStrictValidate | kSecCSCheckAllArchitectures;
+  if (osMinorVersion > 8) {
+    flags |= kSecCSCheckNestedCode;
   }
 
-  flags = sFlags;
-  return Status(0, "ok");
+  if (!hash_resources) {
+    flags |= kSecCSDoNotValidateResources;
+  }
+
+  return static_cast<SecCSFlags>(flags);
 }
 
 // Generate a signature for a single file.
-void genSignatureForFile(const std::string& path, QueryData& results) {
+void genSignatureForFile(const std::string& path,
+                         QueryData& results,
+                         bool hash_resources) {
   Row r;
   OSStatus result;
 
   // Defaults
+  r["hash_resources"] = (hash_resources ? "1" : "0");
```

Comment:
Hrm, seems like extra, but I'd use the `INTEGER` macro with the numeric constants.